docs/multi_process.md

```diff
@@ -0,0 +1,127 @@
+# Using JAX in multi-host and multi-process environments
+
+(test-ref)=
+## Introduction
+
+This guide explains how to use JAX in environments such as Cloud TPU pods where
+accelerators are spread across multiple CPU hosts or JAX processes. We’ll refer
+to these as “multi-process” environments.
+
+This guide specifically focuses on how to use collective communication
+operations (e.g. {func}`jax.lax.psum`) in multi-process settings, although other
+communication methods may be useful too depending on your use case (e.g. RPC,
+[mpi4jax](https://github.com/mpi4jax/mpi4jax)). If you’re not already familiar
+with JAX’s collective operations, we recommend starting with the
+{doc}`/jax-101/06-parallelism` notebook. Note that an important feature of
+multi-process environments is direct communication links between accelerators,
+e.g. the high-speed interconnects for Cloud TPUs or NCCL for GPUs. This is what
```

Comment:
Suggestion @skye :

- Maybe some users may not be aware of what NCCL stands for. Perhaps you could spell out the acronym like `NVIDIA Collective Communication Library [thing] (NCCL) for GPUs` 🤷  (that's what NCCL stands for, right? 😃 )

Reply:
The whole doc is amazing and well-written. I'm learning a lot here 👍 Thanks @skye

Reply:
As a compromise, I added a link to https://developer.nvidia.com/nccl where they define the acronym, because the expanded acronym is painfully long :P

Thanks, glad you're finding it useful :D